fractions/javaee/jaxrs/src/main/java/org/wildfly/swarm/jaxrs/internal/JAXRSArchiveImpl.java

```diff
@@ -99,8 +112,62 @@ private static boolean hasApplicationPathAnnotation(Archive<?> archive) {
         return false;
     }
 
+    private static boolean hasApplicationServletMapping(Archive<?> archive) {
+        Node webXmlNode = archive.get(PATH_WEB_XML);
+        if (webXmlNode != null) {
+            return hasApplicationServletMapping(webXmlNode.getAsset());
+        }
+        return false;
+    }
+
+    private static boolean hasApplicationServletMapping(Asset webXmlAsset) {
```

Comment:
Well, my point is that you don't have to use `WebXmlAsset` to specify the `web.xml` in a test (which is by the way the case of the EAP quickstart mentioned in the issue description.. if I'm not wrong). You could use e.g. a `FileAsset` or `StringAsset`. And in that case the only reliable way to detect the servlet mapping is to parse the "real" `web.xml` contents.

A reasonable compromise might be to check `asset instanceof WebXmlAsset || asset instanceof WebAppDescriptor` and use a new accessor or `WebAppDescriptor.getAllServletMapping()` respectively but fallback to XML parsing otherwise.